conference.js

```diff
@@ -713,10 +753,46 @@ export default {
     },
     /**
      * Simulates toolbar button click for audio mute. Used by shortcuts and API.
-     * @param mute true for mute and false for unmute.
+     * @param {boolean} mute true for mute and false for unmute.
+     * @param {boolean} [showUI] when set to false will not display any error
+     * dialogs in case of media permissions error.
      */
-    muteAudio(mute) {
-        muteLocalAudio(mute);
+    muteAudio(mute, showUI = true) {
+        // Not ready to modify track's state yet
+        if (!this._localTracksInitialized) {
+            this.audioMuted = mute;
+            return;
+        } else if (localAudio && localAudio.isMuted() === mute) {
+            // NO-OP
+            return;
+        }
+
+        const maybeShowErrorDialog = (error) => {
+            if (showUI) {
+                APP.UI.showDeviceErrorDialog(error, null);
+            }
+        };
+
+        if (!localAudio && this.audioMuted && !mute) {
+            createLocalTracks({ devices: ['audio'] }, false)
+                .then(([audioTrack]) => audioTrack)
+                .catch(error => {
+                    maybeShowErrorDialog(error);
+
+                    // Rollback the audio muted status by using null track
+                    return null;
+                })
+                .then(audioTrack => this.useAudioStream(audioTrack));
```

Comment:
Is there a reason to move useAudioStream from the first .then( to the last one?

The bug that @yanas is describing actually comes from here. When an error occurs it hits the catch and the second then.